File: connectors/src/main/java/com/binance4j/margin/dto/InterestRate.java
package com.binance4j.margin.dto;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * A loan interest rate.
 * 
 * @property asset             Related coin.
 * @property dailyInterestRate Daily interest rate.
 * @property timestamp         Timestamp of the applied interest rate in ms.
 * @property vipLevel          Account VIP level.
 */
@ApiModel("A loan interest rate.")
data class InterestRate(
@ApiModelProperty("Related coin.")
@JsonProperty("asset") var asset: String = "",
@ApiModelProperty("Daily interest rate.")
@JsonProperty("dailyInterestRate") var dailyInterestRate: String = "",
@ApiModelProperty("Timestamp of the applied interest rate in ms.")
@JsonProperty("timestamp") var timestamp: Long = 0L,
@ApiModelProperty("Account VIP level.") @JsonProperty("vipLevel") var vipLevel:Int = 0)
{
}

File: connectors/src/main/kotlin/com/binance4j/margin/dto/InterestRate.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.margin.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * A loan interest rate.
 *
 * @property asset             Related coin.
 * @property dailyInterestRate Daily interest rate.
 * @property timestamp         Timestamp of the applied interest rate in ms.
 * @property vipLevel          Account VIP level.
 */
@ApiModel("A loan interest rate.")
data class InterestRate(
	@ApiModelProperty("Related coin.")
	val asset: String,
	@ApiModelProperty("Daily interest rate.")
	val dailyInterestRate: String,
	@ApiModelProperty("Timestamp of the applied interest rate in ms.")
	val timestamp: Long,
	@ApiModelProperty("Account VIP level.") var vipLevel: Int
)
